### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by titaniumbones** [larstvei/ox-gfm#22]
Hi Lars,

I'd love to reuse some of your code for another markdown flavour.  (cf. https://github.com/kaushalmodi/ox-hugo/pull/8, which maybe you're already seeing due to an @ mention).  Do you have thoughts about the license under which it's released? If you'd be willing to give it an open-soure license, then GPL would make it easiest to use with Emacs.  It's easy to add the license to your repo if you choose to do it: 

https://help.github.com/articles/adding-a-license-to-a-repository/#including-an-open-source-license-in-your-repository

Thank you!

**Comment by larstvei** [comment]
It is already GPL. Check the header of the `ox-gfm.el`.

**Comment by larstvei** [comment]
Hi!

I was on my phone yesterday and tried to respond to this quickly, but it might have come off a bit snarky.

Currently, I haven't signed the papers to assign the copyright to the FSF. To me, it's not that important, because it is under a free license anyway. If it were to be added to Emacs, then I'd be glad to sign over the rights, but I don't think that will happen.

Anyways, feel free to use all the code you'd like!

**Comment by larstvei** [comment]
I checked the license now and noticed that the license references Emacs and not the more generic "this program" which it was supposed to.

Updated it. Thanks for raising the issue, as it did uncover a problem with the license!

**Comment by kaushalmodi** [comment]
> If it were to be added to Emacs, then I'd be glad to sign over the rights, but I don't think that will happen.

@titaniumbones and I are working on [this project][1], and that is taking a lot of help from `ox-gfm` :) 

I plan to merge `ox-hugo` into Org source, which eventually gets merged into Emacs source on timely basis (for instance, Org 9.0.9 is getting merged into Emacs master this Sunday ([ref][2])). So anything inside Org becomes a part of Emacs and needs the FSF assignment.

It will be great if you'd consider assigning the copyright to FSF.

[1]: https://github.com/kaushalmodi/ox-hugo 
[2]: http://lists.gnu.org/archive/html/emacs-devel/2017-06/msg00664.html

**Comment by larstvei** [comment]
I see. I'll send a request to FSF!

**Comment by larstvei** [comment]
Hi, I won't be able to send the form before Monday, hope that is OK.

Even though I have the copyright, there are still six other people who have contributed some code to this. Is that going to be a problem?

**Comment by kaushalmodi** [comment]
> Hi, I won't be able to send the form before Monday, hope that is OK.

Ah, I didn't mean to suggest that you finished the copyright assignment process by this Sunday. Only the maintenance branch of Org gets merged into Emacs master.

Here are the steps in detail:

1. Complete the copyright assignment
2. Email `ox-gfm.el` to the org mailing list for review
3. After the review and updates, commit to Org `master` branch.
4. Once that branch matures, that gets merged to the stable or `maint` branch.
5. That then get merged into Emacs `master`.

So there is not need for you to hurry at the moment :)

> There are still six other people who have contributed some code to this. Is that going to be a problem?

It depends on the amount (number of lines) of contribution. If the contribution is about 15 lines or less, they do not need to do the copyright assignment ([ref][1]). So looking at the [contributors][2] of `ox-gfm`, that leaves @scottoasis ([commits][3]) and @Christopher-Bui ([commits][4]).

@scottoasis, @Christopher-Bui Would you guys like to assign the copyright to FSF so that `ox-gfm` becomes completely eligible for merger into Org source?

[1]: https://www.gnu.org/prep/maintain/html_node/Legally-Significant.html
[2]: https://github.com/larstvei/ox-gfm/graphs/contributors
[3]: https://github.com/larstvei/ox-gfm/commits?author=scottoasis
[4]: https://github.com/larstvei/ox-gfm/commits?author=Christopher-Bui

**Comment by larstvei** [comment]
Ok.

`ox-gfm.el` used to be in the `/contrib` directory of Org, but was removed after #14. If it's going into Emacs, then this repo needs to be depricated, and the formula should probably be removed from melpa.

I checked the file with blame, and it seems like @jkrmr also has committed more than 15 lines of code (but does not show up in "contributors" because the commit is not connected to his GitHub account).

**Comment by kaushalmodi** [comment]
> ox-gfm.el used to be in the /contrib directory of Org, but was removed after #14.

The contrib dir is as good as Melpa. One doesn't need FSF copyright assignment to put stuff there. As a result, code from `contrib/` also cannot be repurposed in Org/Emacs source.

> If it's going into Emacs, then this repo needs to be depricated, 

That does not need to happen, you just set multiple git remotes. See how the development of packages like [`swiper`][1] and [`diff-hl`][2] happen.. they are on GitHub, Melpa and GNU Elpa.

> and the formula should probably be removed from melpa.

That is up to you. You can keep it there too, and probably remove *after* the package gets into stable version of Org.

> I checked the file with blame, and it seems like @jkrmr also has committed more than 15 lines of code (but does not show up in "contributors" because the commit is not connected to his GitHub account).

OK.

---

In the worst case, i.e. if the other contributors do not respond or do not wish to assign copyright to FSF, you would need to remove their commits and re-implement those (if you wish) using your own code. But as I said, there is no hurry at the moment. 

[1]: https://github.com/abo-abo/swiper
[2]: https://github.com/dgutov/diff-hl

**Comment by larstvei** [comment]
Ok, great. This repo is not really a high priority project for me (as I have other things going on), but I really want you guys to be able to use whatever you want from it. So I'll try to make that as easy as possible for you.

**Comment by Christopher-Bui** [comment]
Sure, just let me know what I need to fill out.

**Comment by kaushalmodi** [comment]
@Christopher-Bui 

You need to send an email to `assign@gnu.org` asking that you'd like to assign your copyrights to FSF. 

This is what I emailed them when I started with my FSF copyright assignment process:

> Hi,
> 
> I visited this site: https://www.gnu.org/prep/maintain/html_node/Copyright-Papers.html#Copyright-Papers
>
> I would like to contribute to GNU Emacs.
> 
> What are the copyright papers that I would need to sign and return back to you?

And the process was over in a week, all via electronic media (email and PDF attachments).

**Comment by larstvei** [comment]
I have signed the papers with the FSF.

If @scottoasis, @Christopher-Bui or @jkrmr have papers in the pipeline then I will hold off a while. If they don't wish to sign then I can start thinking of rewriting their parts so that it can be commited Org mode. Does that sound OK @kaushalmodi?

**Comment by jkrmr** [comment]
@larstvei I've signed them too 👍

**Comment by larstvei** [comment]
Great!

**Comment by kaushalmodi** [comment]
@larstvei Awesome! Now the next step would be to ask on the Org mailing list the next step for making ox-gfm.el a part of Org. My guess would be adding documentation to the Org Info manual.

**Comment by larstvei** [comment]
Cool! I just noticed that @Christopher-Bui has only committed to the README and that it probably be a part of the Org mode version (seeing that it uses Org Info manual instead), so we don't need to wait for his papers to come through.

I think we do have to wait for @scottoasis sign the papers, or if that doesn't happen, rewrite the features he has contributed.
